sdk/identity/identity/src/credentials/clientSecretCredential.browser.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import qs from "qs";
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { TokenCredentialOptions, IdentityClient } from "../client/identityClient";
+import { createSpan } from "../util/tracing";
+import { SpanStatusCode } from "@azure/core-tracing";
+import { credentialLogger, formatError, formatSuccess } from "../util/logging";
+import { getIdentityTokenEndpointSuffix } from "../util/identityTokenEndpoint";
+
+const logger = credentialLogger("ClientSecretCredential");
+
+/**
+ * Enables authentication to Azure Active Directory using a client secret
+ * that was generated for an App Registration.  More information on how
+ * to configure a client secret can be found here:
+ *
+ * https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-configure-app-access-web-apis#add-credentials-to-your-web-application
+ *
+ */
+export class ClientSecretCredential implements TokenCredential {
```

Comment:
Proabably better if you can add a comment mentioning the difference with node.

Reply:
Yes, especially if we want to discourage this to our users. But, you can also do that in a separate PR. I would say log an issue to document the concerns of using this in the browser.

Reply:
But the difference is node is more useful for us as maintainers of this package and less so for customers. So, I would keep that out of the jsdoc/public docs, but somewhere in this file

Reply:
This PR by itself is not very useful if our tests can't use them. Harsha found that browser bundles are broken on Identity v2 when we build v2 with Karma (my manual tests with Webpack did not break). So I had to add some changes that Harsha and I found that would fix browser bundles with Karma.

I also added this comment to this file:

```
// This credential is exported on browser bundles for development purposes.
// For this credential to work in browsers, browsers would need to have security features disabled.
// Please do not disable your browser security features.
```

Harsha also made an issue to track that we need to write browser tests for Identity. Which is going to be mocks of the `InteractiveBrowserCredential` internals, but will help us catch issues with Karma. https://github.com/Azure/azure-sdk-for-js/issues/14913